benchmarks/graphics_pipeline/main.cpp

```diff
@@ -701,49 +454,296 @@ void ProjApp::Setup()
+
+        // Populate vertex buffers for single spheres
+        for (uint32_t j = 0; j < sphereVertexCount; ++j) {
+            TriMeshVertexData vertexData = {};
+            mesh.GetVertexData(j, &vertexData);
+
+            TriMeshVertexDataCompressed vertexDataCompressed;
+            vertexDataCompressed.position = half3(glm::packHalf1x16(vertexData.position.x), glm::packHalf1x16(vertexData.position.y), glm::packHalf1x16(vertexData.position.z));
+            vertexDataCompressed.texCoord = half2(glm::packHalf1x16(vertexData.texCoord.x), glm::packHalf1x16(vertexData.texCoord.y));
+            vertexDataCompressed.normal   = i8vec4(MapFloatToInt8(vertexData.normal.x), MapFloatToInt8(vertexData.normal.y), MapFloatToInt8(vertexData.normal.z), MapFloatToInt8(1.0f));
+            vertexDataCompressed.tangent  = i8vec4(MapFloatToInt8(vertexData.tangent.x), MapFloatToInt8(vertexData.tangent.y), MapFloatToInt8(vertexData.tangent.z), MapFloatToInt8(vertexData.tangent.a));
+
+            lowPrecisionInterleavedSingleSphere.AppendVertexData(vertexDataCompressed);
+            lowPrecisionPositionPlanarSingleSphere.AppendVertexData(vertexDataCompressed);
+            highPrecisionInterleavedSingleSphere.AppendVertexData(vertexData);
+            highPrecisionPositionPlanarSingleSphere.AppendVertexData(vertexData);
+        }
+
+        // Copy single sphere vertex buffers into full buffers, since the non-position vertex buffer data is repeated.
+        RepeatGeometryNonPositionVertexData(lowPrecisionInterleavedSingleSphere, kMaxSphereInstanceCount, lowPrecisionInterleaved);
+        RepeatGeometryNonPositionVertexData(lowPrecisionPositionPlanarSingleSphere, kMaxSphereInstanceCount, lowPrecisionPositionPlanar);
+        RepeatGeometryNonPositionVertexData(highPrecisionInterleavedSingleSphere, kMaxSphereInstanceCount, highPrecisionInterleaved);
+        RepeatGeometryNonPositionVertexData(highPrecisionPositionPlanarSingleSphere, kMaxSphereInstanceCount, highPrecisionPositionPlanar);
+
+        // Get position buffers and prepare to edit in place
+        Geometry::Buffer* lowInterleavedPositionBufferPtr  = lowPrecisionInterleaved.GetVertexBuffer(0);
+        Geometry::Buffer* lowPlanarPositionBufferPtr       = lowPrecisionPositionPlanar.GetVertexBuffer(0);
+        Geometry::Buffer* highInterleavedPositionBufferPtr = highPrecisionInterleaved.GetVertexBuffer(0);
+        Geometry::Buffer* highPlanarPositionBufferPtr      = highPrecisionPositionPlanar.GetVertexBuffer(0);
+
+        // Resize empty Position Planar vertex buffers
+        lowPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * lowPlanarPositionBufferPtr->GetElementSize());
+        highPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * highPlanarPositionBufferPtr->GetElementSize());
+
+        // Iterate through the full vertex buffers, changing position data and appending indices.
+        //
+        // i : sphere index
```

Comment:
I think variables like i and j can be used for simple short loops. Here I think it's more beneficial to name them what they are: sphereIndex, vertexIndex, etc in the code below. sphereIndex loop is quite long, and you can't even easily search to see where it is used.

Reply:
Fixed, kept i/j/k for diagram in comments but changed the code implementation